File: Exercism/javascript/nth-prime/nth-prime.js
'use strict'

export class Prime {
  constructor () {
  }
  
  * gen_primes () {
    yield 2
    const primes = []
    for (let n = 3; ;n += 2) {
      if (primes.every(p => n % p != 0)) {
        primes.push(n)
        yield n
      }
    }
  }

  nth (number) {
    if (!Number.isInteger(number) || number < 1) {
      throw Error('Prime is not possible')
    }
    const iter = this.gen_primes()
    for (let count = 1; count < number; count += 1) {
      iter.next()
    }
    return iter.next().value
  }
}

File: Exercism/python/nth-prime/nth_prime.py
from itertools import count, islice


def gen_primes() -> iter:
    """Generates primes in order"""
    primes = set()
    for n in count(2):
        if all(n % p != 0 for p in primes):
            yield n
            primes.add(n)


def prime(nth: int) -> int:
    """Returns the nth prime"""
    return next(islice(gen_primes(), nth - 1, None))


def prime_range(n: int) -> int:
    """Returns a list of the first n primes"""
    return list(islice(gen_primes(), n))
